api/phpunit.xml.dist

```diff
@@ -2,50 +2,45 @@
 
 <!-- https://phpunit.readthedocs.io/en/latest/configuration.html -->
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.4/phpunit.xsd"
     backupGlobals="false"
     colors="true"
     bootstrap="tests/bootstrap.php"
-    convertDeprecationsToExceptions="false"
+    cacheDirectory=".phpunit.cache"
 >
-    <php>
-        <ini name="display_errors" value="1" />
-        <ini name="error_reporting" value="-1" />
-        <server name="APP_ENV" value="test" force="true" />
-        <server name="SHELL_VERBOSITY" value="-1" />
-        <server name="SYMFONY_PHPUNIT_REMOVE" value="" />
-        <server name="SYMFONY_PHPUNIT_VERSION" value="9.6.13" />
-        <env name="SYMFONY_DEPRECATIONS_HELPER" value="quiet[]=indirect&amp;max[total]=999999" />
-        <!-- if "Other deprecation notices" ever gets annoying: 
+  <php>
+    <ini name="display_errors" value="1"/>
+    <ini name="error_reporting" value="-1"/>
+    <server name="APP_ENV" value="test" force="true"/>
+    <server name="SHELL_VERBOSITY" value="-1"/>
+    <server name="SYMFONY_PHPUNIT_REMOVE" value=""/>
+    <server name="SYMFONY_PHPUNIT_VERSION" value="10.4.2"/>
+    <env name="SYMFONY_DEPRECATIONS_HELPER" value="quiet[]=indirect&amp;max[total]=999999"/>
+    <!-- if "Other deprecation notices" ever gets annoying: 
           <env name="SYMFONY_DEPRECATIONS_HELPER"
         value="quiet[]=indirect&amp;quiet[]=other&amp;max[total]=999999"/> -->
-    </php>
+  </php>
 
-    <testsuites>
-        <testsuite name="Project Test Suite">
-            <directory>tests</directory>
-        </testsuite>
-    </testsuites>
+  <testsuites>
+    <testsuite name="Project Test Suite">
+      <directory>tests</directory>
+    </testsuite>
+  </testsuites>
 
-    <listeners>
-        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener" />
```

Comment:
This is not needed anymore?

Reply:
Listeners are not supported anymore with phpunit > 10
symfony-phpunit-bridge does not yet have a replacement, but it works now.
https://phpunit.de/announcements/phpunit-10.html

Reply:
Ah I see. Doesn't this mean that we loose all the Symfony depreciation warnings then?

Reply:
It seems not:
```
Remaining direct deprecation notices (17)

  15x: Since api-platform/core 3.1: The "openapiContext" option is deprecated, use "openapi" instead.
    15x in ResponseSnapshotTest::testOpenApiSpecMatchesSnapshot from App\Tests\Api\SnapshotTests

  1x: Passing an instance of DateTimeImmutable is deprecated, use Doctrine\DBAL\Types\DateTimeImmutableType::convertToDatabaseValue() instead. (DateTimeType.php:51 called by UTCDateTimeType.php:41, https://github.com/doctrine/dbal/pull/6017, package doctrine/dbal)
    1x in UTCDateTimeTypeTest::testDateTimeImmutableConvertsToDatabaseValue from App\Tests\Types\Doctrine

  1x: Passing an instance of DateTimeImmutable is deprecated, use Doctrine\DBAL\Types\DateImmutableType::convertToDatabaseValue() instead. (DateType.php:51 called by UTCDateType.php:26, https://github.com/doctrine/dbal/pull/6017, package doctrine/dbal)
    1x in UTCDateTypeTest::testDateTimeImmutableConvertsToDatabaseValue from App\Tests\Types\Doctrine

Remaining indirect deprecation notices (9250)

Other deprecation notices (7380)

  1477x: Since symfony/framework-bundle 6.4: The "annotations.cached_reader" service is deprecated without replacement.
    41x in EndpointQueryCountTest::testNumberOfQueriesDidNotChange from App\Tests\Api\SnapshotTests
    8x in CreateColumnLayoutTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in UpdateColumnLayoutTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in CreateMaterialNodeTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    8x in UpdateMaterialNodeTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    ...

  1477x: Since symfony/framework-bundle 6.4: The "annotations.reader" service is deprecated without replacement.
    41x in EndpointQueryCountTest::testNumberOfQueriesDidNotChange from App\Tests\Api\SnapshotTests
    8x in CreateColumnLayoutTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in UpdateColumnLayoutTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in CreateMaterialNodeTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    8x in UpdateMaterialNodeTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    ...

  1477x: Since symfony/framework-bundle 6.4: The "annotations.cache_adapter" service is deprecated without replacement.
    41x in EndpointQueryCountTest::testNumberOfQueriesDidNotChange from App\Tests\Api\SnapshotTests
    8x in CreateColumnLayoutTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in UpdateColumnLayoutTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in CreateMaterialNodeTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    8x in UpdateMaterialNodeTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    ...

  1475x: Since symfony/serializer 6.4: Passing a "Doctrine\Common\Annotations\PsrCachedReader" instance as argument 1 to "Symfony\Component\Serializer\Mapping\Loader\AttributeLoader::__construct()" is deprecated, pass null or omit the parameter instead.
    41x in EndpointQueryCountTest::testNumberOfQueriesDidNotChange from App\Tests\Api\SnapshotTests
    8x in CreateColumnLayoutTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in UpdateColumnLayoutTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in CreateMaterialNodeTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    8x in UpdateMaterialNodeTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    ...

  1471x: Since symfony/validator 6.4: Method "Symfony\Component\Validator\ValidatorBuilder::setDoctrineAnnotationReader()" is deprecated without replacement.
    41x in EndpointQueryCountTest::testNumberOfQueriesDidNotChange from App\Tests\Api\SnapshotTests
    8x in CreateColumnLayoutTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in UpdateColumnLayoutTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\ColumnLayout
    8x in CreateMaterialNodeTest::testCreateRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    8x in UpdateMaterialNodeTest::testPatchRejectsParentsWhichDontSupportChildren from App\Tests\Api\ContentNodes\MaterialNode
    ...

  1x: $typeBuilder argument of TypeConverter implementing "ApiPlatform\GraphQl\Type\TypeBuilderInterface" is deprecated since API Platform 3.1. It has to implement "ApiPlatform\GraphQl\Type\TypeBuilderEnumInterface" instead.
    1x in InvitationGraphQLTest::testFindInvitationWhenNotLoggedIn from App\Tests\Api\Invitations

  1x: $typeBuilder argument of FieldsBuilder implementing "ApiPlatform\GraphQl\Type\TypeBuilderInterface" is deprecated since API Platform 3.1. It has to implement "ApiPlatform\GraphQl\Type\TypeBuilderEnumInterface" instead.
    1x in InvitationGraphQLTest::testFindInvitationWhenNotLoggedIn from App\Tests\Api\Invitations

  1x: $fieldsBuilder argument of SchemaBuilder implementing "ApiPlatform\GraphQl\Type\FieldsBuilderInterface" is deprecated since API Platform 3.1. It has to implement "ApiPlatform\GraphQl\Type\FieldsBuilderEnumInterface" instead.
    1x in InvitationGraphQLTest::testFindInvitationWhenNotLoggedIn from App\Tests\Api\Invitations
```
we still have to fix them :wink: